### PR TITLE
Added elrepo

### PR DESCRIPTION
**Pull request by SergiuCip**
Hi, 

I have just added elrepo module. I hope you and others who uses this module, including myself will find it useful.

Thanks,
Cip

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -7,6 +7,36 @@
   $epel_includepkgs = absent
   $epel_exclude = absent
 
+  # ELREPO Settings.
+  # ELREPO has four repos, "main", "testing", "kernel" and "extras".
+  $elrepo_main_url = "http://elrepo.org/linux/elrepo/el${::operatingsystemmajrelease}/${::architecture}"
```
Comment (jeffsheltren) [inline manifests/params.pp]: I don't like setting both 'baseurl' and 'mirrorlist' in yum repo configs. I think a better approach here would be to set all the ```_url``` variables to ```absent```, and only leave the mirrorlist setting for each repo.

File: manifests/params.pp [hunk 2]
```diff
@@ -7,6 +7,36 @@
...
+  $elrepo_main_mirrors = "http://elrepo.org/mirrors-elrepo.el${::operatingsystemmajrelease}"
+  $elrepo_main_enabled = '1'
+  $elrepo_main_gpgcheck = '1'
+  $elrepo_main_includepkgs = absent
+  $elrepo_main_exclude = absent
+
+  $elrepo_testing_url = "http://elrepo.org/linux/testing/el${::operatingsystemmajrelease}/${::architecture}}"
+  $elrepo_testing_mirrors = "http://elrepo.org/mirrors-elrepo.el${::operatingsystemmajrelease}"
+  $elrepo_testing_enabled = '1'
```
Comment (jeffsheltren) [inline manifests/params.pp]: By default ElRepo only enables the main repo and leaves testing, kernel, and extras disabled by default. We should do the same here.

File: files/etc/pki/rpm-gpg/RPM-GPG-KEY-elrepo [hunk 3]
```diff
@@ -0,0 +1,30 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (jeffsheltren) [inline files/etc/pki/rpm-gpg/RPM-GPG-KEY-elrepo]: I would prefer to keep the original filename for this GPG Key. Although I don't particularly like the way they named it, I'd rather re-use what they already have in place.  They use ```RPM-GPG-KEY-elrepo.org```

**Comment by jeffsheltren** [comment]
Thanks for the PR.  I'd love to include ElRepo in the module!  This is very close, but I've made a few comments on changes I would like to see before I merge it.

File: manifests/params.pp (hunk 1 above)
Comment (SergiuCip) [inline manifests/params.pp]: I understand.

**Comment by SergiuCip** [comment]
You're more than welcome and I'm glad I could help. All the feedback it is appreciated as helps me learn more. Thanks a lot.

**Comment by jeffsheltren** [comment]
Hi @SergiuCip, are you able to make the above changes and test this out? Otherwise, I'll give it a shot when I have a chance.

**Comment by SergiuCip** [comment]
Hi @jeffsheltren I have just made the changes that I believe complies with your feedback, however, I am not really sure if that's the right way of dealing params.pp file. 

Please feel free to re-do it the way it should be.

Thank you.

**Comment by jeffsheltren** [comment]
I did a bit of cleanup, but this got merged in, thanks.

**Comment by jeffsheltren** [comment]
This is available in version 0.6.0 on Puppet Forge

**Pull request closed without merging**